Allows non ancestor queries inside a transaction

step76 opened the pull request:
Hello,

As stated [here](https://cloud.google.com/datastore/docs/firestore-or-datastore#in_datastore_mode), the new Firestore in Datastore mode allows executing non-ancestor queries inside a transaction. The library enforces the previous behaviour, so these changes allow leveraging the new feature using the legacy bundle.

This could be useful for many people still using the legacy bundle.

ludoch commented:
Anyway we can add a test via the local appengine datastore dev emulator?
Is it only a client library change or do we need to change the dev emulation as well?

step76 commented:
> Anyway we can add a test via the local appengine datastore dev emulator? Is it only a client library change or do we need to change the dev emulation as well?

I also changed the emulator. The change was pretty trivial. There was a point where the code assumed there was an ancestor in a transaction context, trusting the checks done in the library, so I changed that check to separate the transaction logic and the ancestor logic. The emulator implementation remained the same.

maigovannon commented:
Maybe I am way off here...but doesn't this change enforce the `firestore in datastore mode` for **all** callers? What about the clients who are using in just plain `datastore` calls? @ludoch

ludoch commented:
Thanks, just so that I have all the context, where "I also changed the emulator...." did you do the change?
Google3? CL?

step76 commented:
> Thanks, just so that I have all the context, where "I also changed the emulator...." did you do the change? Google3? CL?

I'm talking about the local dev server contained in this project. One of the three files I've changed in this PR i about the local dev server.

ludoch reviewed (approved):
Approval, but still need to wait for extra internal tests done inside Google code base.

ludoch commented:
After minor internal adaptation, this PR is now back to gh repository.
I will initiate a Cloud CLI update so the release can be in new Cloud CLI Tuesday in a week from now.
